File: Medium/Minimum Falling Path Sum/minimum-falling-path-sum.py
class Solution:
    def minFallingPathSum(self, matrix: List[List[int]]) -> int:

        for i in range(1, len(matrix)):
            for j in range(len(matrix[0])):
                a = matrix[i-1][j]
                b = matrix[i-1][max(0, j-1)]
                c = matrix[i-1][min(j+1, len(matrix[0]) - 1)]

                matrix[i][j] += min(a, b, c)

        return min(matrix[-1])

File: Medium/Minimum Falling Path Sum/minimum-falling-path-sum.cpp
class Solution {
public:
    int minFallingPathSum(vector<vector<int>>& matrix) {

        int ms = matrix[0].size();

        for(int i = 1; i < ms; ++i)
        {
            for(int j = 0; j < ms; ++j)
            {
                int a = matrix[i-1][j];
                int b = matrix[i-1][max(0, j - 1)];
                int c = matrix[i-1][min(ms - 1, j + 1)];

                matrix[i][j] += min(a, min(b, c));
            }
        }
        
        return *min_element(
            matrix[ms - 1].begin(),
            matrix[ms - 1].end()
        );
    }
};
